src/impl/realm_coordinator.cpp

```diff
@@ -465,6 +470,67 @@ void RealmCoordinator::pin_version(VersionID versionid)
     }
 }
 
+#if REALM_ENABLE_SYNC
+void RealmCoordinator::register_partial_sync_query(Realm& realm, Query query, std::string subscription_name)
+#else 
+void RealmCoordinator::register_partial_sync_query(Realm& realm, Query, std::string)
+#endif
+{
+    if (!realm.is_partial())
+        throw std::logic_error("A partial sync query can only be registered in a partially synced Realm");
+
+#if REALM_ENABLE_SYNC
+    std::string object_class = query.get_table()->get_name().substr(6);
+
+    if (realm.schema().find(object_class) == realm.schema().end())
+        throw std::logic_error("A partial sync query can only be registered for a type that exists in the Realm's schema");
+
+    std::string key = subscription_name;
+    std::string serialized_query = "FIXME"; // query.get_description();
```

Comment:
The `query.get_description()` method exists since core 3.2.0 so you could already use it here, just that the text it will output is not conforming to the parser's standard.

Reply:
I'm not sure that has ever been released with Sync though, at least it didn't seem so last I tried. But I'll look into it.